src/town_gui.cpp

```diff
@@ -185,6 +195,33 @@ struct TownAuthorityWindow : Window {
 		}
 	}
 
+	/** Draws the contents of the actions panel. May re-initialise window to resize panel, if the list does not fit. */
+	void DrawActions()
+	{
+		NWidgetBase* nwid = this->GetWidget<NWidgetBase>(WID_TA_COMMAND_LIST);
+		uint left = nwid->pos_x + WD_FRAMERECT_LEFT;
+		uint right = nwid->pos_x + nwid->current_x - 1 - WD_FRAMERECT_RIGHT;
+		uint y = nwid->pos_y + WD_FRAMERECT_TOP;
+
+		DrawString(left + WD_FRAMERECT_LEFT, right - WD_FRAMERECT_RIGHT, y, STR_LOCAL_AUTHORITY_ACTIONS_TITLE);
+		y += FONT_HEIGHT_NORMAL;
+
+		/* Draw list of actions */
+		for (int i = 0; i < TACT_COUNT; i++) {
+			/* Don't show actions if disabled in settings. */
+			if (!HasBit(this->enabled_actions, i)) continue;
+
+			/* Set colour of action based on ability to execute and if selected. */
+			TextColour action_colour = TC_GREY | TC_NO_SHADE;
+			if (HasBit(this->available_actions, i)) action_colour = TC_ORANGE;
+			if (this->sel_index == i) action_colour = TC_WHITE;
+
+			DrawString(left + WD_FRAMERECT_LEFT, right - WD_FRAMERECT_RIGHT, y,
+				STR_LOCAL_AUTHORITY_ACTION_SMALL_ADVERTISING_CAMPAIGN + i, action_colour);
```

Comment:
codestyle: no need for a linebreak here (or, if there is, we use 2 tabs for continuation)